### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ig1na**

### CODE REVIEW

File: docs/custom-rules/README.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
+# Custom Rules
+
+## How to add custom rules
+
+In order to add your custom rules, first you have to install our project as a devDependency with :
+
+`npm i -DE {{ packageName }}`
+
+Then you need to create a folder with the name of your choice, containing another folder named 'rules'.
+In this 'rules' folder, you can create your first rule. You need to write them using TypeScript as we use decorators to register rules into our application.
```
Comment (EmmanuelDemey) [inline docs/custom-rules/README.md]: You are able to use decorator without TypeScript. https://babeljs.io/docs/en/babel-plugin-proposal-decorators

File: docs/custom-rules/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+export class Rule2 {...}
+```
+
+When you're finished writing your rules, run our application using the custom flag :
+
+```bash
+bin/run -c=path/to/custom-folder path/to/your/project
+```
+
+<script>
```
Comment (EmmanuelDemey) [inline docs/custom-rules/README.md]: Why do have this javascript snippet here ? I do not understand

File: src/logger/index.ts [hunk 3]
```diff
@@ -1,4 +1,7 @@
 import { createLogger, format, transports } from 'winston';
+import * as Transport from 'winston-transport';
+import winston = require('winston');
+// import { ui } from '../utils/prompt';
```
Comment (EmmanuelDemey) [inline src/logger/index.ts]: You must remove this commented line

File: src/logger/index.ts [hunk 4]
```diff
@@ -7,6 +10,7 @@ import { createLogger, format, transports } from 'winston';
  */
 export const logger = createLogger({
   level: 'info',
+  // silent: true,
```
Comment (EmmanuelDemey) [inline src/logger/index.ts]: idem

File: src/rules/DEV/JS-TS/FRONT/nginx/index.ts [hunk 5]
```diff
@@ -1,48 +1,46 @@
...
+          })
+          .then(
+            () => {
+              logger.info(
+                'Succesfully copied nginx.conf file to config folder.',
+              );
+              resolve();
+            },
+            err => {
+              if (err.message.endsWith('already exists')) {
```
Comment (EmmanuelDemey) [inline src/rules/DEV/JS-TS/FRONT/nginx/index.ts]: We can maybe use error code instead of a `endsWith` ?

File: src/utils/file-utils/index.ts [hunk 6]
```diff
@@ -47,3 +49,19 @@ export function existsPaths(...paths: string[]): Promise<boolean> {
     return exist.includes(true);
   });
 }
+
+export async function myCopy(from: string, to: string) {
```
Comment (EmmanuelDemey) [inline src/utils/file-utils/index.ts]: Is it useful to name it `myCopy` ? Why not `copy` ?

File: docs/custom-rules/README.md (hunk 2 above)
Comment (ig1na) [reply docs/custom-rules/README.md]: It is a vue script to get the packageName variable

File: src/rules/DEV/JS-TS/FRONT/nginx/index.ts (hunk 5 above)
Comment (ig1na) [reply src/rules/DEV/JS-TS/FRONT/nginx/index.ts]: That's what I wanted to do but sadly there is no error code in the thrown error...

**Pull request merged**